File: unpack.R
#!/bin/Rscript
library(ptools)
library(plyr)
source("library.R")

# unpack the CSV files into an R data file, with a column "exp_type"
# for the experiment type
main <- function(outfile="data.RData",
                 filenames=dir(".", "*.series_trials.csv"), ...) {
  filenames <- c(filenames, ...)
  types <- sub('.*?([a-zA_Z]*)_series_trials.*', '\\1', filenames)
  data <- mdply(data.frame(f=I(filenames), t=types),
                function(f, t) cbind(read.csv(f, stringsAsFactors=FALSE), exp_type=t))
  data$f <- NULL
  data$t <- NULL
  save(data, file=outfile)
}

run_as_command()

File: unpack.m
function unpack(varargin)
% loads all the data into structs.
% output variables are S, a struct containing all the experiment
% types, and 'data', in which experiment type is another column.

if isempty(varargin)
    d = dir('*_trials.csv');
    filenames = {d.name};
else
    filenames = varargin;
end

data = cellfun(@load_data_set, filenames, 'UniformOutput', 0);
function d = load_data_set(filename)
    groupname = regexp(filename, '([a-zA-Z])*_series_trials.*', 'tokens');
    d = dataset(read_csv(filename));
    d.exp_type = repmat(groupname{1}, size(d,1),1);
    %This flip is because the knob recorded values with a flipped mapping.
    d.folded_response_with_carrier = ~d.folded_response_with_carrier;
    d.abs_response_cw = ~d.abs_response_cw;
end

data = dsvertcat(1, data{:});

save(fullfile(fileparts(mfilename('fullpath')), 'data.mat'), 'data')

end
